### PR TITLE
Added Mui, Dayjs, and config date provider

### PR DESCRIPTION
**Pull request by fairnakub**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,13 +3,19 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@date-io/dayjs": "^2.13.1",
```
Comment (jakkapanksb) [inline package.json:6]: Not being used?

File: src/App.tsx [hunk 2]
```diff
@@ -3,30 +3,39 @@ import { QueryClient, QueryClientProvider } from "react-query";
 import { ReactQueryDevtools } from "react-query/devtools";
 import "./App.css";
 import logo from "./logo.svg";
+import { theme } from "./theme";
+import { CustomLocalizationProvider } from "./common/components/CustomLocalizationProvider";
+import { CssBaseline } from "@mui/material";
+import { ThemeProvider } from "@mui/private-theming";
```
Comment (jakkapanksb) [inline src/App.tsx]: I think it's better to import all from `@mui/material`
```suggestion
import { ThemeProvider } from "@mui/material";
```

Comment (jakkapanksb) [inline src/App.tsx]: I think it's better to import all from `@mui/material`
```suggestion
import { ThemeProvider } from "@mui/material";
```

File: src/common/components/CustomLocalizationProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      }));
+    },
+  };
+}
+
+export const CustomLocalizationProvider: FC = (props) => {
+  // TODO: Add a selector to get the current language
+  // TODO: Update "locale" prop in <LocalizationProvider/>
+  return (
+    <LocalizationProvider dateAdapter={DateAdapter as any} locale={"th"}>
```
Comment (jakkapanksb) [inline src/common/components/CustomLocalizationProvider.tsx]: Can we avoid using `any`?

Comment (jakkapanksb) [inline src/common/components/CustomLocalizationProvider.tsx]: Can we avoid using `any`?

File: src/common/components/CustomLocalizationProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    getWeekdays: (): { charAt: () => string }[] => {
+      const start = adapter.dayjs().startOf("week");
+      return [0, 1, 2, 3, 4, 5, 6].map((diff) => ({
+        charAt: () => adapter.formatByString(start.add(diff, "day"), "dd"),
+      }));
+    },
+  };
+}
+
+export const CustomLocalizationProvider: FC = (props) => {
```
Comment (jakkapanksb) [inline src/common/components/CustomLocalizationProvider.tsx]: Should we name it `DateProvider`? I think it is more clear for internal commu

Comment (jakkapanksb) [inline src/common/components/CustomLocalizationProvider.tsx]: Should we name it `DateProvider`? I think it is more clear for internal commu

File: package.json (hunk 1 above)
Comment (fairnakub) [reply package.json:6]: This is probably used somehow by the Mui Date Picker krub
ref: https://mui.com/x/react-date-pickers/getting-started/

File: src/common/components/CustomLocalizationProvider.tsx (hunk 4 above)
Comment (fairnakub) [reply src/common/components/CustomLocalizationProvider.tsx]: Krub. Agreed.

Comment (fairnakub) [reply src/common/components/CustomLocalizationProvider.tsx]: Fixed in [2eb5ff5](https://github.com/jakkapanksb/fe-template/pull/2/commits/2eb5ff55c91537e07fbda099a597649d7b993a71)

File: src/App.tsx (hunk 2 above)
Comment (fairnakub) [reply src/App.tsx]: Fixed in [ed39ed3](https://github.com/jakkapanksb/fe-template/pull/2/commits/ed39ed39a9b044ecb0d61489d706aa42a3d7ef47)

**Comment by jakkapanksb** [comment]
LGTM

**Pull request merged**